app/controllers/questions_controller.rb

```diff
@@ -1,28 +1,52 @@
 class QuestionsController < ApplicationController
 
-  before_action :set_question, only: %i[show edit update destroy]
+  before_action :set_question, only: %i[edit update destroy]
 
   def index
     @question = Question.new
-    @questions = Question.order(id: :desc)
+    @questions = Question.order(id: :desc).includes(:user)
   end
 
   def show
-
+    @question = Question.find(params[:id])
+    @answers = @question.answers
+    @answer = Answer.new
   end
 
   def create
-    question = Question.create!(question_params)
-    redirect_to question, notice: "質問を投稿しました"
+    # @question = current_user.questions.create!(question_params)
+    # redirect_to question, notice: "質問を投稿しました"
+    @question = Question.new(question_params)
+    @question.user_id = current_user.id
```

Comment:
こちらと同じ内容ですので，変更する必要はありません。

```suggestion
    @question = current_user.questions.new(question_params)
```